x/incentive/keeper/bond.go

```diff
@@ -0,0 +1,101 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/umee-network/umee/v4/x/incentive"
+)
+
+// getAllBonded gets the total amount of a uToken bonded to an account across all three unbonding tiers
+func (k Keeper) getAllBonded(ctx sdk.Context, addr sdk.AccAddress, denom string) sdk.Coin {
+	bonded := sdk.NewCoin(denom, sdk.ZeroInt())
+	for _, tier := range []incentive.BondTier{incentive.BondTierLong, incentive.BondTierMiddle, incentive.BondTierShort} {
+		bonded = bonded.Add(k.GetBonded(ctx, addr, denom, tier))
+	}
+	return bonded
+}
+
+// accountBonds gets the total bonded and unbonding for a given account, as well as a list of ongoing unbondings,
+// for a single bond tier and uToken denom. Using incentive.BondTierUnspecified returns all tiers instead.
+// It reduces the returned bonded amount by completed unbondings without actually clearing those unbondings in state,
+// so it is safe for queries and any parts of Msg functions which are not intended to alter state.
+func (k Keeper) accountBonds(ctx sdk.Context, addr sdk.AccAddress, denom string, tier incentive.BondTier) (
+	bonded sdk.Coin, unbonding sdk.Coin, unbondings []incentive.Unbonding,
+) {
+	bonded = sdk.NewCoin(denom, sdk.ZeroInt())
+	unbonding = sdk.NewCoin(denom, sdk.ZeroInt())
+	unbondings = []incentive.Unbonding{}
+
+	// sum all bonded tokens for this denom, for the specified tier or across all tiers
+	for _, t := range []incentive.BondTier{incentive.BondTierLong, incentive.BondTierMiddle, incentive.BondTierShort} {
+		if tier == incentive.BondTierUnspecified || t == tier {
+			bonded = bonded.Add(k.GetBonded(ctx, addr, denom, t))
+		}
+	}
+
+	time := k.GetLastRewardsTime(ctx)
+	storedUnbondings := k.GetUnbondings(ctx, addr)
+	for _, u := range storedUnbondings {
+		if tier == incentive.BondTierUnspecified || u.Tier == uint32(tier) {
```

Comment:
Maybe tire should be a byte? We won't have more than 256 tires

Reply:
It's being kept as a byte in KVStore, but I seem to have found uint32 easier to use in proto. I'll go back and check to see if that still makes sense though